modules/apps/remote-app/remote-app-client-js/src/main/resources/META-INF/resources/js/index.ts

```diff
@@ -323,6 +323,17 @@ function serializeBody(body: BodyInit | null): StructuredClonable {
 	} else {
 		// warn about unserializable body type (eg. FormData, search params,
 		// readable stream),
+		if (body instanceof FormData) {
+			const formData: any = {};
+
+			body.forEach((value: any, key: string) => {
+				formData[key] = value;
+			});
+
+			return {
+				formData,
```

Comment:
I think this property should have an intentionally ugly name to make it obvious that we're passing this data via a "side channel", and one which shouldn't collide with a "legit" object key; eg.

```js
const serializedFormData: {[key: string]: File | string} = {};

body.forEach((value: File | string, key: string) => {
    serializedFormData[key] = value;
});

return {
  __FORM_DATA__: serializedFormData,
};
```

Reply:
:+1: